src/radical/pilot/agent/rm/loadleveler.py

```diff
@@ -625,7 +623,7 @@ def _bgq_create_sub_block_shape_table(self, shape_str):
 
                 # Calculate the number of nodes for the current shape
                 from operator import mul
-                num_nodes = reduce(mul, filter(lambda length: length != 0, sub_block_shape.values()))
+                num_nodes = reduce(mul, [length for length in sub_block_shape.values() if length != 0])
 
```

Comment:
Whats wrong with filter actually?  It should not be slower than list comprehension really?

Reply:
Ugliness and readability?! https://stackoverflow.com/questions/3013449/list-comprehension-vs-lambda-filter no difference really, just preference. Big fun of list comprehension here :P 💃

Reply:
Yeah, both ugliness and readability are very much in the eye of the beholder it seems ;-)

Reply:
pylint jumped with this

Reply:
It threw an error on `filter`?  Ugh.  While some people might not like this, its a totally valid construct which has advantages in some use cases, compared to list comprehension.  Unless we understand why pylint complains here, this is a warning I would ignore...

Reply:
It is an anti-pattern: https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html

Reply:
I would suggest not blindly converting filter and map to LC.  In the cases in this file its fine - but lets not remove valid, legal, well formatted and performant code just because some anti-pattern guide says so, at least not without a better motivation...

I will leave the current changed in place.